bittide/src/Bittide/Transceiver/Comma.hs

```diff
@@ -0,0 +1,49 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+module Bittide.Transceiver.Comma where
+
+import Clash.Explicit.Prelude
+
+import Bittide.Arithmetic.Time (IndexMs)
+import Bittide.SharedTypes (Bytes, Byte)
+import Clash.Class.Counter (Counter(countSuccOverflow))
+
+-- | Number of milliseconds to generate commas for
+defTimeout :: SNat 1
+defTimeout = SNat
+
+-- | Generate commas (transceiver alignment symbols) for a number of milliseconds
+generator ::
+  forall ms nBytes dom .
+  ( KnownDomain dom
+  , KnownNat nBytes
+  , 1 <= ms ) =>
+  SNat ms ->
+  Clock dom ->
+  Reset dom ->
+  -- | (comma, tx control)
+  ( Signal dom (Maybe (Bytes nBytes))
+  , Signal dom (BitVector nBytes)
+  )
+generator _nCycles@SNat clk rst = unbundle $
+  mux
+    (counter .==. pure maxBound)
+    (pure (Nothing, 0))
+    (pure (Just commas, maxBound))
+ where
+  comma :: Byte
+  comma = 0xbc
+
+  commas :: Bytes nBytes
+  commas = pack (repeat comma)
+
+  counter =
+    register
+      clk rst enableGen
+      (0 :: Index ms, 0 :: IndexMs dom 1)
```

Comment:
Is there any particular reason for this split, or in other words, can't we simply offer the user the same time interface as for https://github.com/bittide/bittide-hardware/blob/703b881e8e468e7916dc01b2f8c49ba559ac4649/bittide/src/Bittide/Arithmetic/Time.hs#L57-L62 which would be much more intuitive IMO, cf. e.g., https://github.com/bittide/bittide-hardware/blob/703b881e8e468e7916dc01b2f8c49ba559ac4649/bittide-instances/src/Bittide/Instances/Hitl/HwCcTopologies.hs#L335

Reply:
The split is there to facilitate runtime reconfiguration of waiting times, because we want to do a parameter sweep at some point to see what the right values for the timeouts / comma generation are. Right now they're chosen (very) arbitrarily.

Similar to: https://github.com/bittide/bittide-hardware/blob/e27f60e3fd216f0643f09dbbe4acdb81f9a5f770/bittide/src/Bittide/Transceiver/ResetManager.hs#L48-L51